nautobot/ipam/models.py

```diff
@@ -596,6 +615,30 @@ def save(self, *args, **kwargs):
             parent = max(supernets, key=operator.attrgetter("prefix_length"))
             self.parent = parent
 
+        # Validate that creation of this prefix does not create an invalid parent/child relationship
+        if self.parent and self.parent.type != constants.PREFIX_ALLOWED_PARENT_TYPES[self.type]:
+            err_msg = f"{self.type.title()} prefixes cannot be created in {self.parent.type} prefixes"
+            raise ValidationError({"type": err_msg})
+
+        invalid_children = Prefix.objects.filter(
+            ~models.Q(id=self.id),
+            ~models.Q(type__in=constants.PREFIX_ALLOWED_CHILD_TYPES[self.type]),
+            parent_id=self.parent_id,
+            prefix_length__gt=self.prefix_length,
+            ip_version=self.ip_version,
+            network__gte=self.network,
+            broadcast__lte=self.broadcast,
+            namespace=self.namespace,
+        )
+
+        if invalid_children.exists():
+            invalid_child_prefixes = [
+                f"{child.cidr_str} ({child.type})" for child in invalid_children.only("network", "prefix_length")
+            ]
+            err_msg = f'Creating prefix "{self.prefix}" in namespace "{self.namespace}" with type "{self.type}" '
+            err_msg += f"would create an invalid parent/child relationship with prefixes {invalid_child_prefixes}"
```

Comment:
Can you change this to a multi-line string instead?
```suggestion
            err_msg = (
                f'Creating prefix "{self.prefix}" in namespace "{self.namespace}" with type "{self.type}" '
                f"would create an invalid parent/child relationship with prefixes {invalid_child_prefixes}"
            )
```